packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js

```diff
@@ -94,6 +114,25 @@ describe( 'JSDoc relation-fixer buildrelations module', () => {
 			'classB'
 		];
 
-		expect( testedDoclet ).to.have.property( 'descendants' ).and.to.eql( expectedProp );
+		expectedProp.forEach( prop => {
+			expect( testedDoclet ).to.have.property( 'descendants' ).and.to.include( prop );
+		} );
+	} );
+
+	it( 'should not allow duplicates in relation arrays', () => {
+		const newDoclets = buildRelations( testDoclets );
+		const testedClass = newDoclets.find( d => d.longname === 'classB' );
+		const testedInterface = newDoclets.find( d => d.longname === 'interfaceB' );
+
+		// removing duplicates
+		const expectedImplementsNested = Array.from( new Set( testedClass.implementsNested ) );
+		const expectedMixesNested = Array.from( new Set( testedClass.mixesNested ) );
+		const expectedAugmentsNested = Array.from( new Set( testedClass.augmentsNested ) );
+		const expectedDescendants = Array.from( new Set( testedInterface.descendants ) );
+
+		expect( testedClass ).to.have.property( 'implementsNested' ).and.to.eql( expectedImplementsNested );
```

Comment:
is `eql` the same as `deep.equal()`? If so, I strongly prefer the second option.